src/main/java/org/jabref/model/openoffice/util/OOResult.java

```diff
@@ -0,0 +1,128 @@
+package org.jabref.model.openoffice.util;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+/*
+ * error cannot be null
+ * result cannot be null
+ *
+ * Void is not allowed for R, use OOVoidResult instead.
+ *
+ * Out of `isPresent()` and `isError()` exactly one is true.
+ */
+public class OOResult<R, E> {
+    private final Optional<R> result;
+    private final Optional<E> error;
+
+    /**
+     * Exactly one of the arguments should be Optional.empty()
+     *
+     * @param result
+     * @param error
+     */
+    private OOResult(Optional<R> result, Optional<E> error) {
+        this.result = result;
+        this.error = error;
+    }
+
+    /**
+     * @param result Null is not allowed.
+     */
+    public static <R, E> OOResult<R, E> ok(R result) {
+        return new OOResult<>(Optional.of(result), Optional.empty());
+    }
+
+    /**
+     * @param error Null is not allowed.
+     */
+    public static <R, E> OOResult<R, E> error(E error) {
+        return new OOResult<>(Optional.empty(), Optional.of(error));
+    }
+
+    /*
+     * Test state
+     */
+
```

Comment:
Please no empty line between a JavaDoc and a method

Reply:
My intent was to express, that it does not belong to the next method,
but the next group (of four). A quick search suggests
[here](https://stackoverflow.com/questions/261876/grouping-javadoc-methods)
that javadoc does not out of the box support documenting a group together.

Am I mistaken? Should I remove the comment?

Reply:
Ah, yeah. It is not common to add comments on groups of methods. In the concrete case, I would remove the comment as "Test state" is not that clear (at least to me). Comment using the other extreme would be: Write a whole paragraph if the comment is really important... - In the context of JabRef, we are writing very little comments as we think that comments might be outdated soon and maintainers do not want update comments.

In the book "Java-by-Comparison", there is a whole example on comments: http://media.pragprog.com/titles/javacomp/examples.pdf.

Reply:
Dropped group comments